### DISCUSSION TITLE
CORS and data URI VTT chapters, possible?

### DISCUSSION THREAD

**Discussion by streamworksaudio** [muxinc/elements#1007] [General]
I am storing VTT chapters in the database and then adding them to the mux-video using PHP to create a data uri (data:text/vtt;base64,....) as a track.  

On Chrome this works as expected and the chapters show, but in Safari I am getting a CORS error.  

CORS can be confusing, but could the error be because src of the video file is from mux.com and the chapters are from a data:uri (my domain).  Is it possible to have a data:url for the chapters (in Safari)?  Makes it easier to manage chapters as text in a database than a file elsewhere (such as our CDN which has our domain as a CNAME).

Cheers

**Comment by luwes** [comment]
could you try setting the `crossorigin` attribute on the `mux-video` element?

I do recall some issues on Safari with playing around with this some time ago
https://codesandbox.io/p/sandbox/mux-player-chapters-json-c96zy4

you could try using a Blob but it does require JS unfortunately.

**Comment by streamworksaudio** [comment]
If I remove 'crossorigin' from the mux-video element then it seems to work.  But thought I needed this attribute.  If I remove it the storyboard.vtt from Mux give a CORS error.

**Reply by streamworksaudio** [reply]
I should add that the CORS error for the storyboard.vtt (when I remove the crossorigin attribute) is in Chrome only.

**Comment by cjpillsbury** [comment]
Hey @streamworksaudio I think this one may sadly be a long lived Chrome/Chromium bug that got closed due to inactivity 💔: https://issues.chromium.org/issues/40411923#comment24

**Reply by streamworksaudio** [reply]
I think this might have been fixed in Chrome (the issue in that link is marked as such).  The problem I am having is with Safari.

For example this URL:

https://streamworksaudio.com/watch/edjnazqqpqoe/

The chapters work in Chrome, but not in Safari.
